### PR TITLE
e2e(upgrades): ordered ICA channel can be closed after upgrade and packet timeout

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #5538

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/interchain_accounts/upgrades_test.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+// initiateChannelUpgrade creates and submits a governance proposal to execute the message to initiate a channel upgrade
+func (s *InterchainAccountsChannelUpgradesTestSuite) initiateChannelUpgrade(ctx context.Context, chain ibc.Chain, wallet ibc.Wallet, portID, channelID string, upgradeFields channeltypes.UpgradeFields) {
+	govModuleAddress, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chain)
+	s.Require().NoError(err)
+	s.Require().NotNil(govModuleAddress)
+
+	msg := channeltypes.NewMsgChannelUpgradeInit(portID, channelID, upgradeFields, govModuleAddress.String())
+	s.ExecuteAndPassGovV1Proposal(ctx, msg, chain, wallet)
+}
```
Comment (DimitrisJim) [inline e2e/tests/interchain_accounts/upgrades_test.go:220]: note: should be able to re-use ones in `upgrades_test.go` in `channel/`. Can open an issue and tackle it later if you want.

**Review by DimitrisJim — APPROVED** [review]
lgtm, left one comment re re-using some functions which you probably also noticed.

**Comment by DimitrisJim** [comment]
also note same e2e is failing for other PRs, seems like a regression someplace (relayer?)

**Comment by crodriguezvega** [comment]
> also note same e2e is failing for other PRs, seems like a regression someplace (relayer?)

I will take a look.

File: e2e/tests/interchain_accounts/upgrades_test.go (hunk 1 above)
Comment (crodriguezvega) [reply e2e/tests/interchain_accounts/upgrades_test.go:220]: I was planning to move the functions to a common place when all the e2e tests are merged.

Comment (DimitrisJim) [reply e2e/tests/interchain_accounts/upgrades_test.go:220]: https://github.com/cosmos/ibc-go/issues/5611 to keep track and assigned you, feel free to unassign if required :muscle:

**Comment by crodriguezvega** [comment]
Looks like in the failing tests the relayer is submitting TRY twice to chain B. I already let Luca know.

**Comment by damiannolan** [comment]
Is this the PR that requires a new relayer image before merging? I see the test is still failing in CI

File: e2e/tests/interchain_accounts/upgrades_test.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+			bz, err := icatypes.SerializeCosmosTx(cdc, []proto.Message{msgSend}, icatypes.EncodingProtobuf)
+			s.Require().NoError(err)
+
+			packetData := icatypes.InterchainAccountPacketData{
+				Type: icatypes.EXECUTE_TX,
+				Data: bz,
+				Memo: "e2e",
+			}
+
+			msgSendTx := controllertypes.NewMsgSendTx(controllerAddress, ibctesting.FirstConnectionID, uint64(1), packetData)
```
Comment (colin-axner) [inline e2e/tests/interchain_accounts/upgrades_test.go:165]:
```suggestion
                        timeout := uint64(1)
			msgSendTx := controllertypes.NewMsgSendTx(controllerAddress, ibctesting.FirstConnectionID, timeout, packetData)
```

just a suggestion, improves readability for me a bit

**Review by colin-axner — APPROVED** [review]
Thank you @crodriguezvega for all your works getting the e2e's to be green!!!! :heart_on_fire:

**Pull request merged**